bin/program.py

```diff
@@ -287,19 +287,35 @@ def _checks(self):
             if self.language == 'cpp':
```

Comment:
can you add `or (self.problem.interactive and isinstance(self, Validator)` and test that?  (I'm not exactly sure whether that `self.problem.interactive` is exactly right.)

Reply:
why not just warn for all validators? I don't see why a normal output validator should be allowed to be non-deterministic but in an interactive problem, we don't want to allow it?

Reply:
ok let's do all and revisit when people complain ;)